### ISSUE TITLE
`contrast-color()` - Contradictory information on contrast guaruntees

### ISSUE THREAD

**Issue by lukewarlow** [mdn/content#41422]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/contrast-color

### What information was incorrect, unhelpful, or incomplete?

Right at the top there's a link saying guarunteed to the WCAG color contrast guidelines.

"The contrast-color() [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [function](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_Value_Functions) takes a [color](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) value and returns a [guaranteed](https://w3c.github.io/wcag/guidelines/22/#contrast-minimum) contrasting color."

But further down there's a big red box saying: "There is no guarantee that the values returned using the contrast-color() function will produce an accessible result."

### What did you expect to see?

I would expect the page to clear demonstrate the guaruntees provided to the author from this function. Per the spec the algorithm is user agent defined and given it can only return white or black I don't think it can actually guaruntee minimum contrast is met in all cases (as noted in the warning note).

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/color_value/contrast-color`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/contrast-color
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/color_value/contrast-color/index.md
* Last commit: https://github.com/mdn/content/commit/0cc9980e3b21c83d1800a428bc402ae1865326b2
* Document last modified: 2025-07-14T14:43:58.000Z

</details>

**Comment by Josh-Cena** [comment]
@OnkarRuikar?

**Comment by OnkarRuikar** [comment]
There are two contrast guidelines: AA (minimum `4.5:1`) and AAA (enhanced `7:1`). The `contrast-color` function adheres to the minimum AA guideline. The minimum `4.5:1` ratio is not capable of producing readable text in [all scenarios](https://webaim.org/resources/contrastchecker/?fcolor=000000&bcolor=2275D3):

![Image](https://github.com/user-attachments/assets/22ab22fa-2eea-41d8-ad51-521f9e96c84f)

Note in the above screenshot that the normal text has 4.5:1 contrast, but it is still somewhat harder to read. 

The contrast is not everything. As per the [specifications](https://drafts.csswg.org/css-color-5/#contrast-color):
> Note: Legibility is a complex topic, and sufficient color contrast is only one piece of the puzzle. Having a color pair with sufficient contrast does not guarantee that the text will be legible, as that also depends on a variety of factors, such as the font, the size of the text, the surrounding colors, etc.

***

> given it can only return white or black I don't think it can actually guaruntee minimum contrast is met in all cases (as noted in the warning note).

No. Remember, you can pass non black and white colors to the function. So, it is all the colors versus black/white. You can play with the above scenario in [the MDN playground](https://developer.mozilla.org/en-US/play?id=MIQnbLsbEymDLWawGecb7iTEYQIULdip30E%2BhSGFtPtdLoHGztsOmHsN0Kqu9g2KaXf%2FRCodjmLa688F) on Safari v26 browser.

In short, the `4.5:1` contrast is guaranteed, but the `4.5:1` contrast doesn't guarantee readable text.

Should we remove the "guarantee" word from the first paragraph?

**Comment by lukewarlow** [comment]
I think the language is contradictory as general understanding of contrasting colour will be, it's perceived with contrast, but it seems it's just mathematically contrasting. So yeah I think rephrasing that first bit to be more explicit about what is and isn't guarunteed would be good.

Also per spec the algorithm is up to the browser so I don't think anything is *guaranteed* in a strict sense.
